odl/solvers/util/callback.py

```diff
@@ -315,6 +315,9 @@ def __init__(self, *args, **kwargs):
         ----------
         display_step : positive int, optional
             Number of iterations between plots. Default: 1
+        saveto : string, optional
```

Comment:
I'm against just giving a directory because the user then has to guess as to how the system is gonna name the files. "Explicit is better than implicit."
I would prefer a ``saveto_fmt`` parameter in which users can specify *both location and file naming pattern*. The canonical example would be ``'/path/to/image/files/image-{:03}.png'`` for less than 1000 files. Link also to [this page]{https://docs.python.org/3/library/string.html#format-examples}, it gives some instructive examples regarding usage of format strings.

Finally we shouldn't raise a `ValueError` for an I/O error. That should be handled by the system. Just let the underlying error propagate.